WordPressLint/src/main/java/org/wordpress/android/lint/MissingNullAnnotationDetector.kt

```diff
@@ -0,0 +1,183 @@
+package org.wordpress.android.lint
+
+import com.android.tools.lint.client.api.UElementHandler
+import com.android.tools.lint.detector.api.Category.Companion.CORRECTNESS
+import com.android.tools.lint.detector.api.Detector
+import com.android.tools.lint.detector.api.Implementation
+import com.android.tools.lint.detector.api.Issue
+import com.android.tools.lint.detector.api.JavaContext
+import com.android.tools.lint.detector.api.LintFix
+import com.android.tools.lint.detector.api.Scope.Companion.JAVA_FILE_SCOPE
+import com.android.tools.lint.detector.api.Severity
+import com.android.tools.lint.detector.api.SourceCodeScanner
+import com.android.tools.lint.detector.api.isJava
+import com.intellij.psi.PsiPrimitiveType
+import org.jetbrains.uast.UAnnotated
+import org.jetbrains.uast.UAnnotationMethod
+import org.jetbrains.uast.UAnonymousClass
+import org.jetbrains.uast.UElement
+import org.jetbrains.uast.UEnumConstant
+import org.jetbrains.uast.UField
+import org.jetbrains.uast.UMethod
+import org.jetbrains.uast.UParameter
+import org.jetbrains.uast.UVariable
+import org.jetbrains.uast.getContainingUClass
+
+class MissingNullAnnotationDetector : Detector(), SourceCodeScanner {
+    override fun getApplicableUastTypes(): List<Class<out UElement>> = listOf(
+            UField::class.java,
+            UMethod::class.java,
+    )
+
+    override fun createUastHandler(context: JavaContext): UElementHandler? = with(context) {
+        if (!isJava(uastFile?.sourcePsi)) {
+            return UElementHandler.NONE
```

Comment:
Suggestion (💡): Btw, now that this is NOT returning `null`, but instead a `UElementHandler.NONE` consider changing this `createUastHandler(...)` methods signature return type to a non null `UElementHandler`. Otherwise AS complains about it: `'createUastHandler' always returns non-null type `

Reply:
Yeah, good idea (I had this in my stash, and forgot to push it)! Resolved with https://github.com/wordpress-mobile/WordPress-Lint-Android/pull/11/commits/bb6d98be145aa9876508d1c53bebf1d0e0032380

Reply:
Awesome, thanks for this update @mkevins ! 💯